hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -473,9 +473,9 @@ case class HoodieFileIndex(spark: SparkSession,
 
   private def validateConfig(): Unit = {
     if (isDataSkippingEnabled && (!isMetadataTableEnabled || !isIndexEnabled)) {
-      logWarning("Data skipping requires both Metadata Table and at least one of Column Stats Index, Record Level Index, or Functional Index" +
+      logWarning("Data skipping requires both Metadata Table and at least one of Column Stats Index, Record Level Index, or Expression Index" +
```

Comment:
Should we fix the log message and condition here? We don't necessarily need colstats to be enabled on the reader side.

Reply:
This is just a warning log. This is basically checking if column stats is available in the table.
I have renamed the function from `isIndexEnabled` to `isIndexAvailable`